src/resource_reader.rs

```diff
@@ -9,6 +9,7 @@ use crate::runtime::TargetRuntime;
 pub struct FileRead {
     pub content: String,
     pub path: String,
+    pub content_ty: Option<String>,
```

Comment:
```suggestion
    pub content_type: Option<String>,
```

Should be an enum

Reply:
On second thoughts, content-type might not be needed. This module doesn't care about the type of the file. The caller might have enough information to make a reasonable guess.